test/dynamo/test_export.py

```diff
@@ -2364,9 +2364,10 @@ def forward(self, x):
 
         with self.assertRaisesRegex(
             torch._dynamo.exc.UserError,
-            "Not all values.*valid.*inferred to be equal to(.*\n)*.*"
+            "Constraints violated .*!(.*\n)*.*"
```

Comment:
Yep, I'm seeing this now:

```
torch._dynamo.exc.UserError: Constraints violated (dim0, dim1)! For more information, run with TORCH_LOGS="+dynamic".
  - The values of dim0 = L['x'].size()[0] must always be related to the values of dim1 = L['x'].size()[1] by dim0 = 2*dim1.
  - Not all values of dim1 = L['x'].size()[1] in the specified range satisfy the generated guard 2 <= L['x'].size()[1] and L['x'].size()[1] <= 5

Suggested fixes:
  dim1 = Dim('dim1', max=5)
  dim0 = 2*dim1
```

compared to this before:

```
torch._dynamo.exc.UserError: Constraints violated (dim1, dim0)! For more information, run with TORCH_LOGS="+dynamic".
  - Not all values of dim0 = L['x'].size()[0] in the specified range 3 <= dim0 <= 10 are valid because dim0 was inferred to be equal to 2*L['x'].size()[1].
  - The values of dim0 = L['x'].size()[0] must always be related to the values of dim1 = L['x'].size()[1] by dim0 = 2*dim1.
  - Not all values of dim1 = L['x'].size()[1] in the specified range satisfy the generated guard 2 <= L['x'].size()[1] and L['x'].size()[1] <= 5

Suggested fixes:
  dim1 = Dim('dim1', max=5)
  dim0 = 2*dim1
```

So one constraint/error message disappears because of this change: https://github.com/pytorch/pytorch/blob/c6c41952f4ce177dbd5bb25fca9556761125329f/torch/fx/experimental/symbolic_shapes.py#L2957

and that was necessary to export this in strict mode:
```
def test_dynamic_dim_1_2(self):
        class Bar(torch.nn.Module):
            def forward(self, x, y):
                return x + y[1:]

        dx = Dim("dx", min=1, max=2)
        ep = export(
            Bar(),
            (torch.randn(2, 2), torch.randn(3, 2)),
            dynamic_shapes=({0: dx, 1: None}, {0: dx+1, 1: None})
        )
```

It seemed that for the input y, dynamo would look at x, do analysis with a Dim(max(2, 1)=2, 2) bound to avoid 0/1 issues, then infer a Dim(3, 3) bound for y, which would conflict with the user-specified Dim(2, 3) constraint and crash.

Reply:
Ah oui. So benign difference then to the error message itself. Btw we should sort the "Constraints violated (<list>)" so that we don't get non-determinism in the order. That can happen separately.